File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.java
class Solution {
    public int maxSubarrayLength(int[] nums, int k) {
        int left = 0;
        Map<Integer, Integer> m = new HashMap<>();
        int answer = 0;

        for (int i = 0; i < nums.length; i++) {
            m.put(nums[i], m.getOrDefault(nums[i], 0) + 1);
            
            while (left != i && k < m.get(nums[i])) {
                m.put(nums[left], m.get(nums[left++]) - 1);
            }
            
            answer = Math.max(answer, i - left + 1);
        }
        
        return answer;
    }
}

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
  int maxSubarrayLength(vector<int>& nums, int k) {
    int l=0;
    map<int,int>m;
    int answer=1;
    for(int i=0;i<nums.size();i++){
      m[nums[i]]++;
      while(l!=i&&k<m[nums[i]]){
        m[nums[l++]]--;
      }
      answer=max(answer,i-l+1);
    }
    return answer;
  }
};